h/views/admin/groups.py

```diff
@@ -101,24 +108,16 @@ def on_success(appstruct):
 
             # Add members to the group. We know that these users exist
             # because that check is part of form schema validation.
-            member_userids = [
-                _userid(username, organization.authority)
-                for username in appstruct["members"]
-            ]
-            group_members_svc.add_members(group, member_userids)
+            member_userids = []
+            for username in appstruct["members"]:
+                member_userids.append(
+                    self.user_svc.fetch(username, organization.authority).userid
+                )
```

Comment:
I think this could be:

```python
member_userids = [self.user_svc.fetch(username, organization.authority).userid
                  for username in appstruct["members"]
```

Which might be a little clearer and more succinct.

Reply:
I love a list comprehension, but this one was getting hard to parse IMHO.